src/stats/inc/LikelihoodBase.h

```diff
@@ -77,10 +85,31 @@ class BaseGaussianLikelihood : public BaseScalarFunction<V, M> {
    * the observational covariance matrix.  In this case, the user need not
    * concern themselves with this parameter as it is handled not in the model
    * evaluation but by the likelihood evaluation.
+   *
+   * By default, to ensure backward compatibility, this method will call the
+   * deprecated
+   * evaluateModel( const V & domainVector, const V * domainDirection,
+   * V & modelOutput, V * gradVector, M * hessianMatrix, V * hessianEffect ).
    */
-  virtual void evaluateModel(const V & domainVector, const V * domainDirection,
-      V & modelOutput, V * gradVector, M * hessianMatrix,
-      V * hessianEffect) const = 0;
+  virtual void evaluateModel(const V & domainVector, V & modelOutput) const
+  { this->evaluateModel(domainVector,NULL,modelOutput,NULL,NULL,NULL); }
+
+  //! Logarithm of the value of the scalar function.
+  /*!
+   * This interface is deprecated. Subclasses should instead implement
+   * double lnValue(const V & domainVector)
+   */
+  virtual double lnValue(const V & domainVector, const V * domainDirection,
+      V * gradVector, M * hessianMatrix, V * hessianEffect) const;
+
+  //! Actual value of the scalar function.
+  virtual double actualValue(const V & domainVector, const V * domainDirection,
+                             V * gradVector, M * hessianMatrix, V * hessianEffect) const
+  { return std::exp(this->lnValue(domainVector, domainDirection, gradVector,
+                                  hessianMatrix, hessianEffect)); }
+
+  //! Logarithm of the value of the scalar function.
+  virtual double lnValue(const V & domainVector) const =0;
```

Comment:
Gotcha.  I spent a little while trying to figure out what the reasoning behind it was before I gave up and just resorted to asking you to explain it to me.  Glad I did :)